src/__tests__/cli.ts

```diff
@@ -1,32 +1,61 @@
 import fs from 'fs';
 import path from 'path';
 import util from 'util';
-import childProcess from 'child_process';
+import { CliArguments } from '..';
 
 const mkdir = util.promisify(fs.mkdir);
 const rmdir = util.promisify(fs.rmdir);
 const writeFile = util.promisify(fs.writeFile);
 const readFile = util.promisify(fs.readFile);
 
-function exec(
-  command: string,
-  options: { cwd: string },
-): Promise<{ exitCode: number | null; stdout: string; stderr?: string }> {
-  return new Promise((resolve) => {
-    try {
-      const buffer = childProcess.execSync(command, options);
-      resolve({
-        exitCode: 0,
-        stdout: buffer.toString(),
-      });
-    } catch (ex) {
-      resolve({
-        exitCode: ex.status,
-        stdout: ex.stdout.toString(),
-        stderr: ex.stderr.toString(),
-      });
-    }
-  });
+async function exec(
+  testProjectDir: string,
+  { init = false, flow = false, update = false }: Partial<CliArguments> = {},
+): Promise<{ exitCode: number | null; stdout: string; stderr: string }> {
+  const originalExit = process.exit;
+  const originalCwd = process.cwd();
+  const originalConsole = {
+    log: console.log,
+    warn: console.warn,
+    error: console.error,
+  };
+
+  try {
+    let exitCode: number | null = null;
+    let stdout = '';
+    let stderr = '';
+
+    const appendStdout = (...args: any[]): void => {
+      stdout += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    const appendStderr = (...args: any[]): void => {
+      stderr += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    console.log = appendStdout;
+    console.warn = appendStdout;
+    console.error = appendStderr;
+
+    process.exit = (code: number): never => {
+      exitCode = exitCode ?? code;
+      return undefined as never;
+    };
+
+    process.chdir(testProjectDir);
+
+    const main = require('..').main as (args: CliArguments) => Promise<void>;
+
+    await main({ init, flow, update });
```

Comment:
Ideally, I would have just done `require('..');` and let `yargs` and everything just work, but I'm not sure if we can replace `process.argv` with the options we want to pass or not.

On top of that, main returns a promise but just importing the file doesn't give me any way to wait for it.  I contemplated just waiting 200ms or however long is reasonable but that seemed brittle to me.